### PR TITLE
Restructure examples, add arg parser and suppress warnings

### PR DESCRIPTION
**Pull request by adityapb**

### CODE REVIEW

**Comment by prabhuramachandran** [comment]
Would be good to fix any user warnings and also update the README.

File: compyle/utils.py [hunk 1]
```diff
@@ -24,3 +25,30 @@ def getsource(obj):
...
+
+class ArgumentParser(argparse.ArgumentParser):
+    '''Standard argument parser for compyle applications.
+    Includes arguments for backend, openmp and use_double
+    '''
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # setup standard arguments
+        self.add_argument(
```
Comment (prabhuramachandran) [inline compyle/utils.py:38]: This should offer a choice.

File: compyle/utils.py [hunk 2]
```diff
@@ -24,3 +25,30 @@ def getsource(obj):
...
+        )
+        self.add_argument(
+            '--use-double', action='store_true', dest='use_double',
+            default=False, help='Use double precision on the GPU.'
+        )
+        self.add_argument(
+            '--suppress-warnings', action='store_true',
+            dest='suppress_warnings',
+            default=False, help='Suppress warnings'
+        )
```
Comment (prabhuramachandran) [inline compyle/utils.py:54]: Perhaps a process options should be made available, or perhaps `set_config_options()` which sets up the configuration based on these options?

File: examples/molecular_dynamics/3D/hoomd_periodic.py [hunk 3]
```diff
@@ -59,4 +65,4 @@ def simulate(num_particles, dt, tf, profile=False):
 
 if __name__ == '__main__':
     import sys
-    print(simulate(int(sys.argv[1]), 0.02, 4., profile=True))
+    print(simulate(int(sys.argv[1]), 0.02, 200., profile=True, log=True))
```
Comment (prabhuramachandran) [inline examples/molecular_dynamics/3D/hoomd_periodic.py:68]: Does turning on profile slow things down?

File: examples/molecular_dynamics/md_nnps.py [hunk 4]
```diff
@@ -133,6 +112,7 @@ def solve(self, t, dt):
     o = p.parse_args()
     get_config().use_openmp = o.openmp
     get_config().use_double = o.use_double
+    get_config().suppress_warnings = o.suppress_warnings
```
Comment (prabhuramachandran) [inline examples/molecular_dynamics/md_nnps.py:115]: This repetitive block will also be fixed, maybe overload `ArgumentParser.parse_args` to automatically handle these default options.

File: examples/molecular_dynamics/3D/hoomd_periodic.py (hunk 3 above)
Comment (adityapb) [reply examples/molecular_dynamics/3D/hoomd_periodic.py:68]: Probably does, but I'm not turning it on in the performance_comparison script

**Comment by prabhuramachandran** [comment]
LGTM!  Thanks.

**Pull request merged**